youtube_dl/extractor/contar.py

```diff
@@ -0,0 +1,301 @@
+            'episode_id': video.get('id'),
+            'duration': int_or_none(video.get('length')),
+            'thumbnail': video.get('posterImage'),
+            'release_year': int_or_none(serie_info.get('year')),
+            # 'timestamp': timestamp,
+            'formats': formats,
+            'subtitles': subtitles,
+        }
+
+        return info
+
+    def _get_serie_info(self, serie_id, headers={}):
+        serie = self._call_api('serie/' + serie_id, serie_id, headers=headers, note='Downloading Serie JSON metadata')
+        return serie
+
+    def _get_season_number(self, serie_info, video_id):
+        for season in serie_info['seasons'].get('data', []):
+            season_number = season.get('name')
+            for episode in season['videos'].get('data', []):
+                if episode.get('id') == video_id:
+                    return season_number
+        return None
+
+    def _get_subtitles(self, subtitles, video_id):
+        subs = {}
+        for sub in subtitles:
+            lang = sub.get('lang').lower()
+            subs[lang] = [{'url': sub.get('url'), 'ext': 'srt'}]
+
+        return subs
+
+    def _get_formats(self, videos, video_id):
+        formats = []
+        for stream in videos:
+            stream_url = stream.get('url')
+            type = stream.get('type')
+            if (type == 'HLS'):
+                formats.extend(self._extract_m3u8_formats(stream_url,
+                                                          video_id, 'mp4', entry_protocol='m3u8_native', m3u8_id='hls',
+                                                          fatal=False))
+            elif (type == 'DASH'):
+                formats.extend(self._extract_mpd_formats(
+                    stream_url, video_id, mpd_id='dash', fatal=False))
+
+        self._sort_formats(formats)
+        return formats
+
+
+class ContarIE(ContarBaseIE):
+
+    _UUID_RE = r'[\da-fA-F]{8}-[\da-fA-F]{4}-[\da-fA-F]{4}-[\da-fA-F]{4}-[\da-fA-F]{12}'
+    _VALID_URL = r'https?://(?:www\.)?cont\.ar/watch/(?P<id>%s)' % _UUID_RE
+    _TEST = {
+        'url': 'https://www.cont.ar/watch/d2815f05-f52f-499f-90d0-5671e9e71ce8',
+        'md5': '72cfee8799d964291433004c557d0b2b',
+        'info_dict': {
+            'id': 'd2815f05-f52f-499f-90d0-5671e9e71ce8',
+            'ext': 'mp4',
+            'title': 'Matilde todos los d\u00edas',
+            'duration': 648,
+            'release_year': 2016,
+            'description': 'Matilde llega a la casa de su abuelo Barbacrespa y en ella accidentalmente rompe un coco m\u00e1gico que la dota de poder.',
+            'season_number': 1,
+            'episode_number': 1,
+            # TODO more properties, either as:
+            # * A value
+            # * MD5 checksum; start the string with md5:
+            # * A regular expression; start the string with re:
+            # * Any Python type (for example int or float)
+        },
+        'params': {
+            'usenetrc': True,
```

Comment:
No.

Reply:
what is the correct way to pass username and password to run the test?

Reply:
By providing username and password in params obviously.